<a href="https://colab.research.google.com/github/pgallino/Organizacion-de-datos-TP1-1C2023/blob/main/spark39.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

In [ ]:
!pip install pyspark
!pip install -U -q PyDrive
!apt install openjdk-8-jdk-headless -qq
import os
os.environ["JAVA_HOME"] = "/usr/lib/jvm/java-8-openjdk-amd64"

In [ ]:
from pydrive.auth import GoogleAuth
from pydrive.drive import GoogleDrive
from google.colab import auth
from oauth2client.client import GoogleCredentials
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark import SparkContext
from pyspark.sql import SQLContext
import pandas as pd

In [ ]:
# create the Spark Session
spark = SparkSession.builder.getOrCreate()

# create the Spark Context
sc = spark.sparkContext

In [ ]:
sqlContext = SQLContext(sc)
df = sqlContext.read.csv('/content/drive/MyDrive/Wikipedia dataset 2c2021/logs.csv', header=True, inferSchema=True)
rdd = df.rdd

39 ¿Cuál es la acción más realizada por usuarios que no están registrados? (⭐⭐)

Me quedo con los que en el campo contributor_id tienen None, porque son los que no están registrados.

In [ ]:
rdd = rdd.filter(lambda x: x.contributor_id == None)

In [ ]:
rdd.take(10)

[Row(item_id='Tomada por [[Usuario:Alexwing|Alexwing]] 6 de Diciembre 2004', timestamp=None, contributor_username=None, contributor_id=None, contributor_ip=None, comment=None, logtype=None, action=None, title=None),
 Row(item_id='©1996/2005 - Rock&Web S.A.', timestamp=None, contributor_username=None, contributor_id=None, contributor_ip=None, comment=None, logtype=None, action=None, title=None),
 Row(item_id='{{GFDL}}', timestamp=None, contributor_username=None, contributor_id=None, contributor_ip=None, comment=None, logtype=None, action=None, title=None),
 Row(item_id='{{GFDL}}', timestamp=None, contributor_username=None, contributor_id=None, contributor_ip=None, comment=None, logtype=None, action=None, title=None),
 Row(item_id='{{GFDL}}', timestamp=None, contributor_username=None, contributor_id=None, contributor_ip=None, comment=None, logtype=None, action=None, title=None),
 Row(item_id='{{GFDL}}', timestamp=None, contributor_username=None, contributor_id=None, contributor_ip=None, 

Armo registros (accion, 1) para contar por key

In [ ]:
rdd = rdd.map(lambda x: (x.action, 1))

In [ ]:
rdd = rdd.reduceByKey(lambda x,y : x+y)

Tomo las acciones más realizadas

In [ ]:
rdd.takeOrdered(10, lambda x: -x[1])

[('create', 123902),
 (None, 8998),
 ('autocreate', 239),
 ('delete', 101),
 ('block', 50),
 ('revision', 36),
 ('protect', 22),
 ('event', 17),
 ('create2', 16),
 (' 173);"" class="""" | ali…»"', 13)]

Por lo que podemos ver que el máximo es create con 123902

El valor es similar al obtenido por pandas en el ejercicio 54.